### ISSUE TITLE
Something wrong with the alpha channel

### ISSUE THREAD

**Issue by liberty752** [nvpro-samples/nvtt_samples#3]
I used the [compress](https://github.com/nvpro-samples/nvtt_samples/tree/main/compress) code to compress a png file. After I got the dds file I put it into photoshop which has got plugin [nvidia-texture-tools-exporter](https://developer.nvidia.com/nvidia-texture-tools-exporter) and found that some alpha channels were wrong.

This is the  original png file. It is round-bordered.
![3011000_best](https://github.com/nvpro-samples/nvtt_samples/assets/1614519/8cc1cdf9-0731-4862-b13c-b3e8c071f9c6)

This is the dds image, which I have transferred to png format with photoshop.
![3011000_best_dds](https://github.com/nvpro-samples/nvtt_samples/assets/1614519/8b1e10c0-9474-4994-af50-e4901422fd96)

the arguments are
-astc_ldr_6x6 3011000_best.png

It looks like that every 6 by 6 block share the same alpha channel.

**Comment by liberty752** [comment]
I just tried to change AlphaMode from AlphaMode_Transparency to AlphaMode_None, although it was a png file with alpha channels. And it was right.

**Comment by NBickford-NV** [comment]
Hi @liberty752 , thanks for reporting this issue! I can confirm this is a bug: it looks like NVTT 3.2.0 and 3.2.1's code for reading the `EncodeFlags_Opaque` flag gave the opposite result of what it should — so images using `AlphaMode_Transparent` or `AlphaMode_Premultiplied` would use the opaque compression code, and images using `AlphaMode_Opaque` would use compressors that supported transparency.

We have a fix for this that will be released in the next version of NVTT (and I'll keep this issue open until we release it); in the meantime, two workarounds are either to specify the opposite alpha mode of the input on the command line (i.e. `-noalpha` for textures with transparency, as you noticed), or to flip the alpha mode in code (which I'll add to the sample here shortly until the new version is released):

```c++
if (alpha_set)
{
  // Workaround for an issue in NVTT 3.2.0 and 3.2.1 where AlphaMode is flipped:
  // https://github.com/nvpro-samples/nvtt_samples/issues/3
  if(30200 <= NVTT_VERSION && NVTT_VERSION <= 30201)
  {
    alphaMode = alpha ? nvtt::AlphaMode_None: nvtt::AlphaMode_Transparency ;
  }
  else
  {
    alphaMode = alpha ? nvtt::AlphaMode_Transparency : nvtt::AlphaMode_None;
  }
}
```

Thanks again!

**Comment by NBickford-NV** [comment]
I've added this workaround in the latest version of `nvtt_compress`, so `-alpha` and `-noalpha` should now work correctly. Thanks again!

**Comment by liberty752** [comment]
Thank you. I'll use the opposite alpha mode until next release.

**Comment by NBickford-NV** [comment]
Hi @liberty752 ! We've just released NVTT 3.2.2, which fixes the swapped alpha mode bug. You can download it from here:

* Windows x86_64: https://developer.nvidia.com/downloads/texture-tools-standalone-app
* Linux x86_64: https://developer.nvidia.com/downloads/texture-tools-linux-x86-64
* Linux AArch64: https://developer.nvidia.com/downloads/texture-tools-linux-aarch64

Thanks again!
